File: frontend/src/services/roleService.js
import api from './api';

const roleService = {
  getAll: (params = {}) => api.get('/roles', { params }),
  getById: (id) => api.get(`/role/${id}`),
  create: (data) => api.post('/role', data),
  update: (id, data) => api.put(`/role/${id}`, data),
  delete: (id) => api.delete(`/role/${id}`),
  assignPermissions: (id, data) => api.post(`/role/${id}/permissions`, data),
  assignMenus: (id, data) => api.post(`/role/${id}/menus`, data),
};

export default roleService;

File: internal/interfaces/role/role_service.go
package interfacerole

import (
	domainrole "safety-riding/internal/domain/role"
	"safety-riding/internal/dto"
	"safety-riding/pkg/filter"
)

type ServiceRoleInterface interface {
	Create(req dto.RoleCreate) (domainrole.Role, error)
	GetByID(id string) (domainrole.Role, error)
	GetByIDWithDetails(id string) (dto.RoleWithDetails, error)
	GetAll(params filter.BaseParams, currentUserRole string) ([]domainrole.Role, int64, error)
	Update(id string, req dto.RoleUpdate) (domainrole.Role, error)
	Delete(id string) error
	AssignPermissions(roleId string, req dto.AssignPermissions) error
	AssignMenus(roleId string, req dto.AssignMenus) error
	GetRolePermissions(roleId string) ([]string, error)
	GetRoleMenus(roleId string) ([]string, error)
}
